src/TesApi.Web/IBatchPoolFactory.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.Batch;
+
+namespace TesApi.Web
+{
+    /// <summary>
+    /// Factory to create BatchPool instances.
+    /// </summary>
+    public interface IBatchPoolFactory
```

Comment:
This factory creates the POCO that manages the batch pool, not the batch pool itself (which lives in Azure's Batch Account), and in fact is called after obtaining `CloudPool`-type objects from Azure (whether in or out of the management-plane).

The reason for the factory is to facilitate DI supplying several of the constructor parameters without hardcoding the calls to the constructors.

Reply:
The need to create multiple instances of `IBatchPool` using different constructors is further evidence that the implementation is too complex. As already suggested, the class should be divided into smaller classes/components. This will allow for simpler DI practices, with a single constructor for each class and removes the need of the factory implementation altogether, which will improve the overall design and maintainability.

Reply:
The only thing complicating the ultimate composition of whatever represents a batch pool is the constructor, unless we use `init` properties and add an "initializer" method (which would force breakage of RO value protections).

I'll look at this, possibly in a new PR, to see where I can simplify more.

Reply:
I looked at this. I should look at it again next month once my brain has had a change of scenery...

Reply:
I've refactored `BatchPool` so there's now only one constructor, and this factory is now greatly simplified. Separately, I've made the batch pool responsible for its own batch job, instead of `BatchScheduler` having to deal directly with it.